### DISCUSSION TITLE
PETSc Updated to 3.20.0

### DISCUSSION THREAD

**Discussion by cticenhour** [idaholab/moose#25891] [News]
The `moose-petsc` package has been updated alongside the repository submodule to version 3.20.0 in #25784.
Subsequently, several conda packages have been incremented and should be updated. To update your MOOSE
packages, please activate your MOOSE conda environment and perform the command:

```
mamba update --all
```

If this command does not succeed in updating the MOOSE packages, this is generally due to a conda
dependency resolution issue, or a pinned package (usually Python). In this case, it is recommended to update
a specific package rather than using the `--all` option. Thus, one can also attempt the command:

```
mamba update moose-dev
```

or replace `moose-dev` with the highest-level MOOSE package installed if performing manual dependency builds 
(e.g., `moose-libmesh`, `moose-petsc`, etc. plus `moose-tools`). The following MOOSE package versions and 
builds outlined in the update prompt should be at least the following:

```
Package                  Version           Build
===================================================
moose-dev                2023.10.23        build_0
moose-libmesh            2023.09.06        build_4
moose-libmesh-vtk        9.2.6             build_5
moose-petsc              3.20.0            build_0
moose-mpich              4.0.2             build_12
moose-wasp               2023.10.06        build_0
moose-peacock            2023.10.23        build_0
moose-tools              2023.08.31        build_0
```

**Comment by jessecarterMOOSE** [comment]
Any notable improvements over previous version?

**Reply by lindsayad** [reply]
A summary of changes can be found at https://petsc.org/release/changes/320/. A couple of additions relevant to MOOSE are:

- fixes to the Least Squares Commutator preconditioner which allows for scalable preconditioning of the incompressible navier-stokes equations
- ability to specify custom norm computations for SLEPc-base eigenvalue calculations

**Reply by maxnezdyur** [reply]
> * fixes to the Least Squares Commutator preconditioner which allows for scalable preconditioning of the incompressible navier-stokes equations

Is there a link to this?

**Reply by lindsayad** [reply]
You can see a fair amount of discussion on the merge request at https://gitlab.com/petsc/petsc/-/merge_requests/6642
